docs/upgrade-govuk-frontend.md

```diff
@@ -0,0 +1,22 @@
+# Keep this gem in sync with the Design System
+
+This gem consumes [`govuk-frontend`](https://github.com/alphagov/govuk-frontend) via [Yarn](https://classic.yarnpkg.com/). [Dependabot](https://github.com/marketplace/dependabot-preview) will let us know when [a new version of `govuk-frontend` is available](https://github.com/alphagov/govuk_publishing_components/pulls?q=is%3Apr+is%3Aopen+label%3Adependencies+bump+govuk-frontend). There are a few checks to be done before merging in a 'Bump `govuk-frontend`' pull request.
+
+1. Read the [release notes](https://github.com/alphagov/govuk-frontend/releases/). GOV.UK Frontend has well-documented release notes that, depending on the release type, might contain the following sections:
+ - Breaking changes (major release)
+ - New features (minor release)
+ - Fixes (patch release)
+
+2. Check if and how the breaking changes affect this gem and GOV.UK applications dependent on this gem. For each breaking change follow the migration steps in the release notes and push the changes in a separate commit on the same Dependabot branch. Make sure to test GOV.UK applications with this branch. On major releases is always a good idea to involve someone from [the Design System team](https://github.com/orgs/alphagov/teams/team-gov-uk-design-system/members) in the review process as they have a thorough understanding of the changes.
+
+3. Check if and how the new feature changes affect this repository. If a new feature is added to a component, check the markup of the component associated with the feature mirrors the one in `govuk-frontend`. We usually benefit from the JavaScript and SCSS updates without requiring any changes, however make sure these are being imported in the component as expected. If a feature was pushed upstream from this gem to the Design System make sure to update the code to use the feature from upstream and remove the unused code in this gem. We decided that we do not bring features in `govuk_publishing_components` unless there is a user need for this (for example it improves the accessibility of a component) or it makes the code more robust (for example it provides better browser support).
```

Comment:
Nitpick, English usage: "we do not bring features in" in should be "into". I think it also works better as "We decided that we **will not** bring features into"